https://github.com/owncloud/core/pull/12140/files?diff=unified
--- a/lib/private/template.php.orig	2015-01-05 17:18:44.533507972 +0000
+++ b/lib/private/template.php	2015-01-05 18:02:07.358508180 +0000
@@ -97,36 +97,28 @@
 	}

 	/**
-	 * Returns the formfactor extension for current formfactor
+	 * Returns the form factor extension for current form factor
 	 */
 	static public function getFormFactorExtension()
 	{
-		if (!\OC::$session) {
-			return '';
-		}
-		// if the formfactor is not yet autodetected do the
-		// autodetection now. For possible formfactors check the
-		// detectFormfactor documentation
-		if (!\OC::$session->exists('formfactor')) {
-			\OC::$session->set('formfactor', self::detectFormfactor());
-		}
 		// allow manual override via GET parameter
 		if(isset($_GET['formfactor'])) {
-			\OC::$session->set('formfactor', $_GET['formfactor']);
+			$formFactor = $_GET['formfactor'];
+		} else {
+			$formFactor = self::detectFormfactor();
 		}
-		$formfactor = \OC::$session->get('formfactor');
-		if($formfactor==='default') {
-			$fext='';
-		}elseif($formfactor==='mobile') {
-			$fext='.mobile';
-		}elseif($formfactor==='tablet') {
-			$fext='.tablet';
-		}elseif($formfactor==='standalone') {
-			$fext='.standalone';
+		if($formFactor==='default') {
+			$formFactorExt='';
+		}elseif($formFactor==='mobile') {
+			$formFactorExt='.mobile';
+		}elseif($formFactor==='tablet') {
+			$formFactorExt='.tablet';
+		}elseif($formFactor==='standalone') {
+			$formFactorExt='.standalone';
 		}else{
-			$fext='';
+			$formFactorExt='';
 		}
-		return $fext;
+		return $formFactorExt;
 	}

 	/**
https://github.com/owncloud/core/pull/11727/files?diff=unified
--- a/lib/private/appframework/middleware/security/securitymiddleware.php.orig	2015-01-05 17:45:33.219507598 +0000
+++ b/lib/private/appframework/middleware/security/securitymiddleware.php	2015-01-05 17:47:26.282675805 +0000
@@ -35,7 +35,7 @@
 use OCP\IRequest;
 use OCP\ILogger;
 use OCP\AppFramework\Controller;
-
+use OCP\Util;

 /**
  * Used to do all the authentication and checking stuff for a controller method
@@ -110,7 +110,8 @@
 				}
 			}
 		}
-
+		// CSRF check - also registers the CSRF token since the session may be closed later
+		Util::callRegister();
 		if(!$this->reflector->hasAnnotation('NoCSRFRequired')) {
 			if(!$this->request->passesCSRFCheck()) {
 				throw new SecurityException('CSRF check failed', Http::STATUS_PRECONDITION_FAILED);
